routes/api.js

```diff
@@ -447,25 +455,37 @@ router.post('/create_account', apiMiddleware(async (req) => {
     account_auths: accountAuths,
     key_auths: [[publicKeys.posting, 1]],
   };
-
-  try {
-    await steem.broadcast.accountCreateWithDelegationAsync(
-      process.env.DELEGATOR_ACTIVE_WIF,
-      process.env.CREATE_ACCOUNT_FEE,
-      process.env.CREATE_ACCOUNT_DELEGATION,
-      process.env.DELEGATOR_USERNAME,
-      username,
-      owner,
-      active,
-      posting,
-      publicKeys.memo,
-      metadata,
-      [],
-    );
-  } catch (cause) {
-    throw new ApiError({ type: 'error_api_create_account', cause });
+  const newCreationHash = await req.db.sequelize.query(
+    'SELECT SQL_NO_CACHE creation_hash FROM users WHERE email = ?',
+    {
+      replacements: [decoded.email],
+      type: req.db.sequelize.QueryTypes.SELECT,
+    },
+  );
+  if (newCreationHash[0].creation_hash === creationHash) {
```

Comment:
@gregory-latinier In async handlers where the pattern is to throw, it is much better to do assertions and throw early  to keep the code readable. Writing this as `if (newHash != oldHash) { throw }` avoids the indentation change and makes it immediately clear that we are not executing anything more if that happens